var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -77,26 +77,45 @@ def url_for_version(self, version):
+                                 'user-diffraction': 'diffraction',
+                                 'user-dpd': 'dpd-react', 'user-drude': 'drude',
+                                 'user-eff': 'eff', 'user-fep': 'fep',
+                                 'user-h5md': 'h5md', 'user-lb': 'latboltz',
+                                 'user-manifold': 'manifold',
+                                 'user-mesodpd': 'dpd-meso', 'user-mesont': 'mesont',
+                                 'user-mgpt': 'mgpt', 'user-mofff': 'mofff',
+                                 'user-netcdf': 'netcdf', 'user-phonon': 'phonon',
+                                 'user-plumed': 'plumed', 'user-ptm': 'ptm',
+                                 'user-qtb': 'qtb', 'user-reaction': 'reaction',
+                                 'user-reaxc': 'reaxff', 'user-sdpd': 'dpd-smooth',
+                                 'user-smd': 'machdyn', 'user-smtbq': 'smtbq',
+                                 'user-sph': 'sph', 'user-tally': 'tally',
+                                 'user-uef': 'uef', 'user-yaff': 'yaff'}
+
     supported_packages = ['asphere', 'body', 'class2', 'colloid', 'compress',
-                          'coreshell', 'dipole', 'granular', 'kspace',
-                          'kokkos', 'latte', 'manybody', 'mc', 'meam', 'misc',
-                          'mliap', 'molecule', 'mpiio', 'opt', 'peri', 'poems',
-                          'python', 'qeq', 'replica', 'rigid', 'shock', 'snap',
-                          'spin', 'srd', 'user-atc', 'user-adios',
-                          'user-awpmd', 'user-bocs', 'user-cgsdk',
-                          'user-colvars', 'user-diffraction', 'user-dpd',
-                          'user-drude', 'user-eff', 'user-fep', 'user-h5md',
-                          'user-lb', 'user-manifold', 'user-meamc',
-                          'user-mesodpd', 'user-mesont', 'user-mgpt',
-                          'user-misc', 'user-mofff', 'user-netcdf', 'user-omp',
-                          'user-phonon', 'user-plumed', 'user-ptm', 'user-qtb',
-                          'user-reaction', 'user-reaxc', 'user-sdpd',
-                          'user-smd', 'user-smtbq', 'user-sph', 'user-tally',
-                          'user-uef', 'user-yaff', 'voronoi']
+                          'coreshell', 'dipole', 'granular',
+                          'kspace', 'kokkos', 'latte', 'manybody', 'mc', 'meam',
+                          'misc', 'molecule', 'mpiio', 'opt', 'peri',
+                          'poems', 'python', 'qeq', 'replica', 'rigid',
+                          'shock', 'spin', 'srd', 'user-meamc', 'user-misc',
+                          'user-omp', 'voronoi']
+    supported_packages.extend(renamed_packages_20210702.keys())
+    supported_packages.extend(renamed_packages_20210702.values())
 
     for pkg in supported_packages:
         variant(pkg, default=False,
                 description='Activate the {0} package'.format(pkg))
```

Comment:
```suggestion
    for pkg in supported_packages:
        if pkg in renamed_packages_20210702.keys():
            variant(pkg, default=False, when='@:20210527',
                    description='Activate the {0} package'.format(pkg))
        if pkg in renamed_packages_20210702.values():
            variant(pkg, default=False, when='@20210702:',
                    description='Activate the {0} package'.format(pkg))
        else:
            variant(pkg, default=False,
                    description='Activate the {0} package'.format(pkg))
```
Using conditional variants might be more suitable here.